e2e/pageobjects/ide/Editor.ts

```diff
@@ -13,44 +13,74 @@ import { DriverHelper } from '../../utils/DriverHelper';
 import { CLASSES } from '../../inversify.types';
 import { TestConstants } from '../../TestConstants';
 import { By, Key } from 'selenium-webdriver';
+import { error } from 'selenium-webdriver';
 
 @injectable()
 export class Editor {
+    private static readonly SUGGESTION_WIDGET_BODY_CSS: string = 'div.visible[widgetId=\'editor.widget.suggestWidget\']';
 
-    private static readonly EDITOR_BODY_CSS: string = '#theia-main-content-panel .lines-content';
-    private static readonly SUGGESTION_WIDGET_BODY_CSS: string = 'div[widgetId=\'editor.widget.suggestWidget\']';
-    private static readonly EDITOR_INTERACTION_СSS: string = '.monaco-editor textarea';
     constructor(@inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) { }
 
-    async waitSuggestionContainer(timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+    public async waitSuggestionContainer(timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
         await this.driverHelper.waitVisibility(By.css(Editor.SUGGESTION_WIDGET_BODY_CSS), timeout);
     }
 
-    async waitSuggestionContainerClosed(attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS, polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
-        await this.driverHelper.waitDisappearance(By.css(Editor.SUGGESTION_WIDGET_BODY_CSS), attempts, polling);
+    public async waitSuggestionContainerClosed(attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
+        polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
+        await this.driverHelper.waitDisappearanceTestWithTimeout(By.css(Editor.SUGGESTION_WIDGET_BODY_CSS));
     }
 
-    async waitSuggestion(suggestionText: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        await this.driverHelper.waitVisibility(By.xpath(this.getSuggestionLineXpathLocator(suggestionText)), timeout);
+    public async waitSuggestion(editorTabTitle: string,
+        suggestionText: string,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+
+        const suggestionLocator: By = By.xpath(this.getSuggestionLineXpathLocator(suggestionText));
+
+        await this.driverHelper.getDriver().wait(async () => {
+            await this.waitSuggestionContainer();
+            try {
+                await this.driverHelper.waitVisibility(suggestionLocator, 5000);
+                return true;
+            } catch (err) {
+                const isTimeoutError: boolean = err instanceof error.TimeoutError;
+                if (!isTimeoutError) {
+                    throw err;
+                }
+
+                await this.pressEscapeButton(editorTabTitle);
+                await this.waitSuggestionContainerClosed();
+                await this.pressControlSpaceCombination(editorTabTitle);
```

Comment:
NOTE 1: If I understand correctly reason of this is - if the suggestion is not shown, test tries to view it by pressing `ctrl + space`. Shouldn't there be another check if the suggestion is shown correctly for that second attempt? 

NOTE 2: Why does pressing ` ctrl + space ` have own method? Why not to call ` performKeyCombination ` directly from here?

Reply:
**Answer 1:** Not exactly. Next method encapsulated in the driver "wait" method, this is a kind of loop, it will repeat until called method returns some value.
```
await this.waitSuggestionContainer();
            try {
                await this.driverHelper.waitVisibility(suggestionLocator, 5000);
                return true;
            } catch (err) {
                const isTimeoutError: boolean = err instanceof error.TimeoutError;
                if (!isTimeoutError) {
                    throw err;
                }

                await this.pressEscapeButton(editorTabTitle);
                await this.waitSuggestionContainerClosed();
                await this.pressControlSpaceCombination(editorTabTitle);
            }
```
**Answer 2:** It looks much better for me to use particular method for invoking suggestion container than repeat each time ```this.performKeyCombination(editorTabTitle, Key.chord(Key.CONTROL, Key.SPACE))```

Reply:
Answer 2: You still need to write a line of code - even performKeyCombination or pressControlSpaceCombination. It is quite weird to have methods that contain only one line this way. And it doesn't seem to be frequent method - only 2 usages. The ` escapeButton ` is called only once.

Reply:
I must say, that I'm not really fond of this approach either. Maybe if these "typical" key combinations would live in separate classes, then it would be ok. But having such methods in Editor class is not very nice IMO.
But I don't want to block this PR because of this.